solr/solrj/src/java/org/apache/solr/common/util/SimpleOrderedMap.java

```diff
@@ -86,6 +91,79 @@ public static SimpleOrderedMap<Object> of() {
    * @return SimpleOrderedMap containing one key-value pair
    */
   public static <T> SimpleOrderedMap<T> of(String name, T val) {
-    return new SimpleOrderedMap<>(List.of(name, val));
+    return new SimpleOrderedMap<T>(List.of(name, val));
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return nvPairs.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(final Object key) {
+    return super.get((String) key) != null;
+  }
+
+  @Override
+  public boolean containsValue(final Object value) {
+    int sz = size();
+    for (int i = 0; i < sz; i++) {
+      T val = getVal(i);
+      if (Objects.equals(value, val)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public T get(final Object key) {
+    return super.get((String) key);
+  }
+
+  @Override
+  public T put(final String key, final T value) {
+    T oldValue = get(key);
+    add(key, value);
```

Comment:
add will not replace the existing value.  This is why that method is called "add" (vs put or set).  I pointed you at NamedList asShallowMap which dealt with this.  This concern also led me to recommend deferring mutability.